File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FollowsResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.entity.Account;
import mastodon4j.entity.share.Response;

/**
 * @author hecateball
 */
public interface FollowsResource {

    /**
     * Following a remote user.
     *
     * @param uri username@domain of the person you want to follow
     * @return the local representation of the followed account, as an Account
     */
    public Response<Account> remoteFollow(String uri);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FollowsResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.follows.FollowsRemoteFollowRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.follows.FollowsRemoteFollowResponse
import kotlin.js.JsExport


@JsExport
interface FollowsResource {

    /**
     * Following a remote user.
     */
    fun remoteFollow(
        request: FollowsRemoteFollowRequest
    ): Response<FollowsRemoteFollowResponse>
}
